pkg/logger/logger.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package logger provides a convenient interface to use to log.
+package logger
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+)
+
+// These are the log levels used by the logger.
+// See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md#what-method-to-use
+const (
+	logLevelWarn  = 1
+	logLevelDebug = 4
+	logLevelTrace = 5
+)
+
+// Wrapper defines a convenient interface to use to log things.
+type Wrapper interface {
+	Info(msg string, keysAndValues ...any)
+	Debug(msg string, keysAndValues ...any)
+	Warn(msg string, keysAndValues ...any)
+	Trace(msg string, keysAndValues ...any)
+	Error(err error, msg string, keysAndValues ...any)
+	WithValues(keysAndValues ...any) *Logger
+	WithName(name string) *Logger
+	GetLogger() logr.Logger
+}
+
+// Logger is a concrete logger using logr underneath.
+type Logger struct {
```

Comment:
Just curious why we are adding a new logger here? And is it going to replace the loggers in other controllers?

Reply:
Yeah, I added this because CAPA does it, but I also think it's a good call @bochengchu  - it feels orthogonal.  I think I will remove it.